doc/users.md

```diff
@@ -1 +1,827 @@
+- region_id from regions tables if type_akses is 2.
+- kios_id from kios tables if type_akses is 4.
+- if you already have a company_id, you don't need to include company_code and company_name in the users table. Because it's will provide additional responsibility to update the users table while there is an update in the companies table.
+- point above also applies to region_code, kios_code, branch_code, and branch_name.
+
+#### Validation
+- name: required and not empty
+- username: required and not empty
+- email: required and not empty
+- type_akses: required and not empty and must in range (1-8)
+- company_id: required and not empty and must be registered in companies if type_akses in range (1-4)
+- region_id: required and not empty and must be registered in regions if type_akses = 2
+- branch_id: required and not empty and must be registered in branches if type_akses in range (3-4)
+- kios_id: required and not empty and must be registered in kios if type_akses = 4
+
+### Scenario Test
+
+#### Case : Negative Case 1
+
+Request Payload : empty
+
+Response HTTP Status Code : 400
+
+Response Payload :
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "name is required",
+    "data": null
+}
+```
+
+#### Case : Negative Case 2
+
+Request Payload :
+```
+{}
+```
+
+Response HTTP Status Code : 400
+
+Response Payload :
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "name is required",
+    "data": null
+}
+```
+
+#### Case : Negative Case 3
+
+Request payload :
+```
+{
+    "name": ""
+}
+```
+
+Response HTTP Status Code : 400
+
+Response Payload :
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "name is empty",
+    "data": null
+}
+```
+
+#### Case : Negative Case 4
+
+Request Payload :
+```
+{
+    "name": "dennis"
+}
+```
+
+Response HTTP Status Code : 400
+
+Response Payload :
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "username is required",
+    "data": null
+}
+```
+
+#### Case : Negative Case 5
+
+Request Payload :
+```
+{
+    "name": "dennis",
+    "username": ""
+}
+```
+ 
+Response HTTP Status Code : 400
+
+Response Payload:
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "username is empty",
+    "data": null
+}
+```
+
+#### Case : Negative Case 6
+
+Request Payload
+```
+{
+    "name": "dennis",
+    "username": "vdennis"
+}
+```
+
+Response HTTP Status Code : 400
+
+Response Payload
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "username already in use",
+    "data": null
+}
+```
+
+#### Case : Negative Case 7
+
+Request Payload
+```
+{
+    "name": "dennis",
+    "username": "vdennis",
+    "password": "Cdcgi2020!"
+}
+```
+
+Response HTTP Status Code : 400
+
+Response Payload
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "email is required",
+    "data": null
+}
+```
+
+#### Case : Negative Case 8
+
+Request Payload
+```
+{
+    "name": "dennis",
+    "username": "vdennis",
+    "password": "Cdcgi2020!",
```

Comment:
no password required. better to generate random password for the first time, because the user will be create by admin, not by user self with registration api.